### DISCUSSION TITLE
I wanted some help with setting up lazygit with enterprise bit bucket

### DISCUSSION THREAD

**Discussion by thesanjithkumar** [jesseduffield/lazygit#3457] [Q&A] [answered]
I use lazygit with github for my personal use in my personal laptop on Arch linux, I loved the lazygit tool and how it can be used to leverage the git productivity, I want to do the same with my work repos in bit bucket but I didn't find any helpful guide on integrating bit bucket with lazygit. Can someone help me with setting up lazygit with Enterprise bit bucket.

**Comment by mark2185** [comment]
What kind of integration did you have in mind?

**Comment by thesanjithkumar** [comment]
My company is using enterprise bit bucket, so wherever I do a git pull it push it asks for a password, so wanted to know if there's a way to use ssh keys or api token can be used with lazygit to go around this password prompt because whenever I try got pull with P key it fails in lazygit.

**Reply by mark2185** [reply]
And does pulling from CLI work without asking for a password? My guess is it asks for a password, and if that's the case you need to setup [bitbucket SSH access](https://support.atlassian.com/bitbucket-cloud/docs/configure-ssh-and-two-step-verification/).

**Reply by thesanjithkumar** [reply]
No it doesn't ask for password it just fails with authentication error, and thanks for the ssh link

**Reply by mark2185** [reply]
Pulling using `git pull` fails with authentication error? That indicates a greater issue, then.

**Reply by thesanjithkumar** [reply]
Ya in my personal device I'm using gh cli to login into my github account and when I use git pull in that device it auto authenticates and performs the pull, but in my work laptop it's not giving any password prompt and tries to search for a locally stored password I think and throws an authentication error message

**Reply by mark2185** [reply]
Okay, SSH access should solve both of those issues and login through `git cli` shouldn't be needed at all.

**Reply by thesanjithkumar** [reply]
ok will give it a try and see, thanks for the suggestion😊

**Comment by thesanjithkumar** [comment, marked as answer]
My organization has blocked all ssh keys, so I was unable to use bit bucket ssh keys for this but I have found a way around. When I use lazygit inside vs code and git bash I get a prompt which asks for my password after entering my password the pull and push works without any issues
